### PR TITLE
feat: allow the port links to be on a customizable domain name

### PR DESCRIPTION
**Pull request by definitelynobody**
Code change to allow #162 to be fixed

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -107,6 +107,12 @@ struct Args {
     #[clap(long)]
     url: auth::Url,
 
+    /// Externally accessible domain name for serving demos.
+    /// This should not be the same as the benefice server URL for security reasons.
+    /// For example: demo.example.com
+    #[clap(long)]
+    demo_domain: String,
```
Comment (rvolosatovs) [inline src/main.rs:114]:
```suggestion
    demo_fqdn: String,
```

**Review by rvolosatovs — COMMENTED** [review]
Small remark

File: templates/idx.html [hunk 2]
```diff
@@ -909,7 +911,7 @@ <h3 class="title is-3">
                     var containerPort = map[1];
                     result +=
                         '<li>Port ' + hostPort + ' -> ' + containerPort +
-                        ' (<a href="https://' + window.location.hostname + ':' + hostPort + '/" target="_blank">link</a>)</li>';
+                        ' (<a href="https://' + demoFqdn + ':' + hostPort + '/" target="_blank">link</a>)</li>';
```
Comment (rvolosatovs) [inline templates/idx.html:914]: Why is this using https? What if the endpoint is plaintext HTTP?
Can we avoid specifying the protocol at all here by dropping the scheme and letting the browser handle that?

Also couldn't we avoid the `<div>` magic and use the `{{ demo_fqdn }}` directly here?

**Review by rvolosatovs — APPROVED** [review]

File: templates/idx.html (hunk 2 above)
Comment (definitelynobody) [reply templates/idx.html:914]: sure i can drop the https

I would prefer not to have the templates generate anything in the script tag

**Pull request merged**